index.bs

```diff
@@ -3051,7 +3051,8 @@ and |session|:
       1. If |match type| is <code>"full"</code> and |node text| [=is=] |search text|,
          [=list/append=] |context node| to |returned nodes|.
 
-      1. Otherwise, [=list/append=] |context node| to |returned nodes|.
+      1. Otherwise, if |match type| is <code>"partial"</code>, [=list/append=]
```

Comment:
Since `match type` can only be “partial” or “full”, this if statement seems redundant. Help me understand where the algorithm fails without the “if” clause?

Reply:
If "match type" is "full" and "node text" is not "search text". For example, if we're searching strictly for "hello" in `<div>hello world</div>`.

Reply:
That’s reasonable. Thanks for the clarification.